README.md

```diff
@@ -1,70 +1,162 @@
 
-```yarn``` must be installed for this project, as it does a better job installing dependendies across platforms.  The dev_setup script above should have done that for you.
+#### Linux/MacOS
 
-### Install modules to the correct locations
+```
+dev_setup.sh
+```
+
+#### Windows
+
+```
+dev_setup.bat
+```
 
-1. ```yarn install``` (always do this after a new clone or checkout)
-2. ```yarn install:pod``` (additional step for macOS if you want to do an iOS build)
+## Running
+
+**Note:** In some cases, these procedures can lead to the error `Failed to load bundle - Could not connect to development server`. In these cases, kill all other react-native processes and try it again.
 
-### Run the app in a simulator
+#### Android (Windows, Linux, macOS)
 
-To run in the Android simulator (Windows, Linux, macOS):
 ```
 npx react-native run-android
 ```
-or (on macOS only)
+
+Device storage can be cleared by long-pressing on the app icon in the simulator, clicking "App info", then "Storage", and lastly, "Clear Storage".
+
+#### iOS (macOS only)
+
 ```
+yarn install:pod ## only needs to be ran once
 npx react-native run-ios
 ```
 
-NOTE: In some cases, these procedures can lead to the error `Failed to load bundle - Could not connect to development server`. In these cases, kill all other react-native processes and try it again.
+Device storage can be cleared by clicking "Hardware" on the system toolbar, and then "Erase all content and settings".
 
-## Contributing
+### Release Builds
 
-Read the [contribution guidelines](CONTRIBUTING.md).
+Generating a release build is an optional step in the development process.
+
+- [Android instructions](https://reactnative.dev/docs/signed-apk-android)
+
+### Debugging
+
+[React Native ships with a debugging tool](https://reactnative.dev/docs/debugging#accessing-the-in-app-developer-menu) that allows you to inspect code, set debugging breakpoints, and more. Each platform has a different command to open the debugger.
+
+**Android:** `⌘D`
+
+**iOS:** `⌘M`
 
+## Testing
+
+Tests are ran automatically through Github actions - PRs are not able to be merged if there are tests that are failing.
+
+### Unit Test
+
+To run the unit tests:
+
+```
+yarn test --watch
+```
+
+To update the snapshots:
+
+```
+yarn update-snapshots
+```
+
+### e2e Test
+
+**Note:** Right now, there is only e2e test support for iOS.
+
+e2e tests are written using [_detox_](https://github.com/wix/Detox). Screenshots of each test run are saved to `e2e/artifacts` for review.
+
+To run the e2e tests:
+
+```
+yarn detox-setup ## only needs to be run once
+yarn build:e2e:ios ## needs to be run after any code change
+yarn test:e2e:iphone{11, -se, 8}
+```
 
+### Manual Device Testing
 
+Mobile devices come in many different shapes and sizes - it is important to test your code on a variety of simulators to ensure it looks correct on all device types.
 
+Before pushing up code, it is recommended to manually test your code on the following devices to verify tha:
```

Comment:
missed part of a sentence here. We also may want to mention a notched device and tablets (if people are using tablets with this at all)

Reply:
Good call.. Pixel 3 XL for Android.

Tablet's I don't think we've made any strides for.

Though perhaps the full detail is better on a wiki.

Reply:
agreed that a more in-depth wiki is better long term. throwing Pixel 3XL and iPhone11 on in the meantime.